Client/Frontend/Browser/Playlist/BrowserViewController+Playlist.swift

```diff
@@ -394,7 +398,33 @@ extension BrowserViewController: PlaylistHelperDelegate, PlaylistFolderSharingHe
   }
   
   // MARK: - PlaylistFolderSharingHelperDelegate
-  func openPlaylistSharingFolder(with playlistId: String) {
-    openPlaylist(tab: nil, item: nil, playbackOffset: 0.0, folderSharingId: playlistId)
+  func openPlaylistSharingFolder(with pageUrl: String) {
+    openPlaylist(tab: nil, item: nil, playbackOffset: 0.0, folderSharingPageUrl: pageUrl)
+  }
+}
+
+extension BrowserViewController {
+  private static var playlistSyncFoldersTimer: Timer?
+  
+  func openPlaylistSettingsMenu() {
+    let playlistSettings = PlaylistSettingsViewController()
+    let navigationController = UINavigationController(rootViewController: playlistSettings)
+    self.present(navigationController, animated: true)
+  }
+  
+  func syncPlaylistFolders() {
+    if Preferences.Playlist.syncSharedFoldersAutomatically.value {
+      BrowserViewController.playlistSyncFoldersTimer?.invalidate()
+      
+      let lastSyncDate = Preferences.Playlist.lastPlaylistFoldersSyncTime.value ?? Date()
+      
+      BrowserViewController.playlistSyncFoldersTimer = Timer(fire: lastSyncDate, interval: 4.hours, repeats: true, block: { _ in
+        Preferences.Playlist.lastPlaylistFoldersSyncTime.value = Date()
+        
+        Task { @MainActor in
+          try await PlaylistManager.syncSharedFolders()
+        }
```

Comment:
The `@MainActor` isolation on this Task created here does nothing

Reply:
Fixed